### PR TITLE
Refactor: Add 'common' package

### PR DESCRIPTION
**Pull request by idillon-sfl**
Just the base of a 'common' package. For now it only includes NotificationType.

### CODE REVIEW

**Comment by idillon-sfl** [comment]
Sorry, I should have paid attention to the CI before putting it into review

File: common/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "target": "es2020",
+    "lib": ["es2020"],
+    "outDir": "out",
```
Comment (deribaucourt) [inline common/tsconfig.json:6]: Have you tested the debug configuration (breakpoints) and .vsix packaging (install and test)?
I didn't know about npm link, it looks useful.

We should use the same way of sharing code between this PR and my WIP https://github.com/savoirfairelinux/vscode-bitbake/pull/41

Let's figure this out on the next meeting :)

File: client/tsconfig.json [hunk 2]
```diff
@@ -13,5 +13,8 @@
     "composite": true
   },
   "include": ["src", ".eslintrc.js"],
-  "exclude": ["node_modules", ".vscode-test"]
+  "exclude": ["node_modules", ".vscode-test"],
+	"references": [
+		{ "path": "../common" }
```
Comment (deribaucourt) [inline client/tsconfig.json:18]: Other ideas to name the common code: common, shared or lib. I don't know which one is more standard.

File: package.json [hunk 3]
```diff
@@ -17,7 +17,8 @@
 		"url": "https://github.com/savoirfairelinux/vscode-bitbake.git"
 	},
 	"scripts": {
-		"postinstall": "cd server && npm install && cd ../client && npm install",
+		"link:common": "cd common && npm link && cd ../client && npm link bitbake-common && cd ../server && npm link bitbake-common",
+		"postinstall": "cd server && npm install && cd ../client && npm install && cd ../common && npm install && cd .. && npm run link:common",
```
Comment (deribaucourt) [inline package.json:21]: You also need to clean the out/ and node_modules folder in npm run clean

File: common/tsconfig.json (hunk 1 above)
Comment (deribaucourt) [reply common/tsconfig.json:6]: You should check:
- [ ]  adding jest __tests__ in the common package
- [ ]  adding a dependency in the common package.json, checking that debugging has it, as well as the .vsix
- [ ]  npm run clean removes all generated files and folders.
- [ ]  generated files are also gitignored
- [ ]  vsix package doesn't contain .ts files
- [ ] breakpoints in the common code get hit when debugging the server and the client
- [ ] vsix works properly once installed

**Review by deribaucourt — CHANGES_REQUESTED** [review]

**Comment by idillon-sfl** [comment]
lets close this

**Pull request closed without merging**